docs/src/main/paradox/binary-compatibility.md

```diff
@@ -0,0 +1,37 @@
+# Binary compatibility
+
+Conflicting versions in transitive dependencies can make upgrading a painful exercise.
+
+To make this easier, starting with version 1.0.0 Akka gRPC provides binary compatibility.
+This means if you use a library that in turn uses Akka gRPC, it should be possible to use
+that library with any newer version of Akka gRPC as well (with the exceptions listed below).
+
+This is especially relevant if you depend on one library that depends on Akka gRPC
+version 'A', and another library that depends on Akka gRPC version 'B': due to
+binary compatibility, you can simply choose the lastest version of Akka gRPC and
+use both libraries with that.
+
+## Limitations
+
+### New features
+
+Features introduced in later versions of Akka gRPC may not work with code generated
+with a previous version of Akka gRPC.
+
+### Deprecations
+
+Binary compatibility can be broken via a deprecation cycle: an API that has been marked deprecated in version `x.y.0`
+may disappear in version `x.(y+1).z`.
+
+### Internal and ApiMayChange API's
+
+Internal API's (designated by the `akka.grpc.internal` package or with the `@InternalApi` annotation) and API's that are still marked `@ApiMayChange` are not guaranteed to remain binary compatible.
+
+Libraries that use such methods may not work in applications that depend on a newer version of Akka gRPC.
+
+## Downstream libraries
+
+We depend on a number of downstream libraries that don't formally maintain binary compatibility,
+such as [ScalaPB](https://scalapb.github.io/) (when generating Scala code) and 
+[grpc-java](https://github.com/grpc/grpc-java/). Updates to those libraries might introduce
+incompatibilities that are outside the scope of Akka gRPC's binary compatibility.
```

Comment:
An upgrade of those might require a new major release of Akka gRPC, right?

Reply:
Yes, it might, but I don't think it must. When grpc-java makes a change that breaks any old code from working with new code, then yes. However, when for instance scalapb makes a change that only breaks code that uses some particular extension, then I think while this is 'strictly binary incompatible' it might make sense decide how we let this impact versioning on a case-by-case basis. What do you think?

Reply:
Yes, that's what I want.
To me "might introduce incompatibilities that are outside the scope of Akka gRPC's binary compatibility" sounds a bit like: "We don't care if these upgrades could break anything".
So maybe write that we will need to decide depending on what those libraries change.